convert2rhel/actions/post_conversion/hostmetering.py

```diff
@@ -0,0 +1,235 @@
+# Copyright(C) 2024 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__metaclass__ = type
+
+
+import logging
+import os
+
+from convert2rhel import actions, systeminfo
+from convert2rhel.pkgmanager import call_yum_cmd
+from convert2rhel.subscription import get_rhsm_facts
+from convert2rhel.systeminfo import system_info
+from convert2rhel.utils import run_subprocess
+
+
+logger = logging.getLogger(__name__)
+
+
+class ConfigureHostMetering(actions.Action):
+    id = "CONFIGURE_HOST_METERING_IF_NEEDED"
+
+    def run(self):
+        """
+        Decide whether to install, enable and start host-metering on the system based on the
+        CONVERT2RHEL_CONFIGURE_HOST_METERING environment variable.
+
+        Behavior can be controlled CONVERT2RHEL_CONFIGURE_HOST_METERING environment variable:
+        - "auto": host-metering will be configured based on the above conditions
+        - env var not present: host-metering will not be configured
+        - "force": forces configuration of host-metering (e.g., even if not running on a hyperscaler)
+        - any other value: Will be ignored and host metering will not be configured.
+
+        :return: True if host-metering is configured successfully, False otherwise (used mainly for testing).
+        :rtype: bool
+        """
+        logger.task("Final: Configure host-metering")
+
+        super(ConfigureHostMetering, self).run()
+
+        self.env_var = os.environ.get("CONVERT2RHEL_CONFIGURE_HOST_METERING", None)
+        if not self._check_env_var_correct():
+            return False
+
+        if system_info.version.major != 7 and self.env_var != "force":
+            logger.info("Skipping host metering configuration. Only supported for RHEL 7.")
+            self.add_message(
+                level="INFO",
+                id="SKIPPED_CONFIGURE_HOST_METERING",
```

Comment:
For reviewers:

When I was testing I found out that when there is used level SKIP, then the action fails, which I think is not what we want here. So I set it to info message and added to the ID `CONFIGURE_HOST_METERING_IF_NEEDED` the `IF_NEEDED` part. I believe this will help to avoid misunderstandings like "Why the host metering was set? I don't need that".

Reply:
@hosekadam so the "SKIP" level isn't supposed to be used in set_result. It's a level for when an action fails and we skip the other ones that depended on it. Reference line - https://github.com/oamg/convert2rhel/blob/81c59f667ac09d9bb1bc148d458db188961effb4/convert2rhel/actions/__init__.py#L53

Reply:
The INFO level is what we want here. For cases where we want to allow the user to skip over a result we use "OVERRIDABLE"